### PR TITLE
LDAP: Adds bind before searching LDAP for non-login cases.

### PR DESCRIPTION
**Pull request by xlson**
**What this PR does / why we need it**:
Adds implicit bind for search except when using LDAP in the Login flow.

**Which issue(s) this PR fixes**:
NONE

**Special notes for your reviewer**:

### CODE REVIEW

File: pkg/services/ldap/ldap.go [hunk 1]
```diff
@@ -380,9 +395,9 @@ func (server *Server) shouldAuthAdmin() bool {
 	return server.Config.BindPassword != ""
 }
 
-// Auth authentificates user in LDAP
-func (server *Server) Auth(username, password string) error {
-	err := server.auth(username, password)
+// UserBind authentificates user in LDAP
```
Comment (markelog) [inline pkg/services/ldap/ldap.go]: You might want to change the description of the method

File: pkg/services/ldap/ldap.go [hunk 2]
```diff
@@ -397,7 +412,7 @@ func (server *Server) Auth(username, password string) error {
 
 // AuthAdmin authentificates LDAP admin user
 func (server *Server) AuthAdmin() error {
```
Comment (markelog) [inline pkg/services/ldap/ldap.go]: Makes sense to remove this method as well?

File: pkg/services/ldap/ldap.go [hunk 3]
```diff
@@ -410,8 +425,8 @@ func (server *Server) AuthAdmin() error {
 	return nil
 }
 
-// auth is helper for several types of LDAP authentification
-func (server *Server) auth(path, password string) error {
+// userBind is helper for several types of LDAP authentification
```
Comment (markelog) [inline pkg/services/ldap/ldap.go]: You might want to change the description of the method

File: pkg/services/ldap/ldap_private_test.go [hunk 4]
```diff
@@ -145,7 +145,7 @@ func TestLDAPPrivateMethods(t *testing.T) {
 	})
 
 	Convey("shouldAuthAdmin()", t, func() {
```
Comment (markelog) [inline pkg/services/ldap/ldap_private_test.go]: It seems this function should be renamed?

File: pkg/services/multildap/testing.go [hunk 5]
```diff
@@ -56,6 +59,11 @@ func (mock *MockLDAP) Close() {
 	mock.closeCalledTimes = mock.closeCalledTimes + 1
 }
 
+func (mock *MockLDAP) Bind() error {
```
Comment (markelog) [inline pkg/services/multildap/testing.go]: Description for this fn?

File: pkg/services/ldap/ldap.go [hunk 6]
```diff
@@ -43,6 +44,20 @@ type Server struct {
 	log        log.Logger
 }
 
+func (server *Server) Bind() error {
```
Comment (markelog) [inline pkg/services/ldap/ldap.go]: Requires method description

**Review by markelog — APPROVED** [review]
Makes sense to me!

**Comment by markelog** [comment]
It seems here, you created a special function for admin and anonymous bind, so I removed the "bug" label

File: pkg/services/ldap/ldap.go (hunk 2 above)
Comment (xlson) [reply pkg/services/ldap/ldap.go]: it does, but I don't want to do that refactoring this late in the game. We'll have to clean that up later

**Pull request merged**